pkg/ccl/backupccl/alter_backup_schedule.go

```diff
@@ -109,20 +108,26 @@ func doAlterBackupSchedules(
 	eval *alterBackupScheduleEval,
 	resultsCh chan<- tree.Datums,
 ) error {
-	s, err := loadSchedules(
-		ctx, p, eval)
+	s, err := loadSchedules(ctx, p, eval)
 	if err != nil {
 		return err
 	}
 
-	// Note that even ADMIN is subject to these restrictions. We expect to
-	// add a finer-grained permissions model soon.
-	if s.fullJob.Owner() != p.User() {
-		return pgerror.Newf(pgcode.InsufficientPrivilege, "only the OWNER of a schedule may alter it")
+	// Check that the user is admin or the owner of the schedules being altered.
+	isAdmin, err := p.UserHasAdminRole(ctx, p.User())
+	if err != nil {
+		return err
 	}
-
-	if s.incJob != nil && s.incJob.Owner() != p.User() {
-		return pgerror.Newf(pgcode.InsufficientPrivilege, "only the OWNER of a schedule may alter it")
+	var isOwner bool
```

Comment:
double check this works right? given this is a permission though, I'd rather not start with isOwner = true and work our way down, but with isOwner = false and build our way up, even if it's slightly more verbose. thoughts?

Reply:
Oh oops, just saw this comment and did exactly what you said you'd rather not to do! Let me rethink this.

Reply:
Check out the latest version, is it better?

Reply:
I like it more, thank you!